src/formatDate.js

```diff
@@ -8,7 +8,57 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromValue = fromFormat[fromFormat.length - 1];
+
+  const toValue = toFormat[toFormat.length - 1];
+
+  const newDate = date.split(fromValue);
+
+  const newArray = [];
+
+  const dateObject = {};
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    dateObject[fromFormat[i]] = newDate[i];
+  }
+
+  const indexOfYear = fromFormat.includes('YYYY')
+    ? fromFormat.indexOf('YYYY')
+    : fromFormat.indexOf('YY');
```

Comment:
The logic for determining the index of the year assumes that either 'YYYY' or 'YY' is present in `fromFormat`. Consider adding validation to handle cases where neither is present, which would cause an error.